### ISSUE TITLE
KeyError: 'color' - Executing bindetect 

### ISSUE THREAD

**Issue by chris-cheshire** [loosolab/TOBIAS#37]
I am getting this error when running on certain datasets/motif combinations:

```
Command error:
  Traceback (most recent call last):
    File "/opt/conda/bin/TOBIAS", line 8, in <module>
      sys.exit(main())
    File "/opt/conda/lib/python3.7/site-packages/tobias/TOBIAS.py", line 154, in main
      args.func(args)rd[0], coor
    File "/opt/conda/lib/python3.7/site-packages/tobias/tools/bindetect.py", line 682, in run_bindetect
      fig = plot_bindetect(comparison_motifs, clustering, [cond1, cond2], args)
    File "/opt/conda/lib/python3.7/site-packages/tobias/tools/bindetect_functions.py", line 640, in plot_bindetect
      ax1.scatter(coord[0], coord[1], color=diff_scores[TF]["color"], s=4.5)
  KeyError: 'color'
```

I believe it traces to this code block:

```
for TF in diff_scores:
		if diff_scores[TF]["change"] < x_min or diff_scores[TF]["change"] > x_max or diff_scores[TF]["log10pvalue"] > y_min:
			diff_scores[TF]["show"] = True
			if diff_scores[TF]["change"] < 0:
				diff_scores[TF]["color"] = "blue"
			elif diff_scores[TF]["change"] > 0:
				diff_scores[TF]["color"] = "red"
		else:
			diff_scores[TF]["show"] = False 
			diff_scores[TF]["color"] = "black"
```

I think the `if` block is not dealing with change == 0 and therefore some cases are not being assigned a `color` dict value.

**Comment by msbentsen** [comment]
Hi, thank you for making me aware of this issue. You are completely right that the color should be set by that code block - what wonders me is how the original "if"-statement is true if the change is 0. Do you have large shifts/differences towards one condition in your data? It seems like the thresholds for change to be either "up" or "down" are both either positive or negative.

Anyway, I will have a look at some changes to make sure this doesn't happen!

**Comment by msbentsen** [comment]
This should be fixed by ` tobias 0.12.4` , but please let me know if this is not the case - thanks!
